problem_builder/mixins.py

```diff
@@ -221,3 +238,57 @@ def student_view_user_state(self, context=None, suffix=''):
         json_result = json.dumps(result, cls=DateTimeEncoder)
 
         return webob.response.Response(body=json_result, content_type='application/json')
+
+
+class StudentViewUserStateResultsTransformerMixin(object):
+    """
+    A convenient way for MentoringBlock and MentoringStepBlock to share
+    student_results transform code.
+    Needs to be used alongside StudentViewUserStateMixin.
+    """
+    def transforms(self):
+        return {
+            'student_results': self.transform_student_results
+        }
+
+    def student_results_transforms(self):
+        """
+        Key (inside `student_results`) to transform function map.
+        Should take
+        """
+        return [
+            ('choices', self.transform_choices),
+            ('tips', self.transform_tips),
+        ]
+
+    def transform_student_results(self, student_results):
+        """
+        Applies `student_results_transforms()` output to current `student_results`
+        """
+        current_student_results = dict(student_results)
+        transformed_student_results = []
+        for question in self.steps:
+            if question.name in current_student_results:
+                question_results = question.get_results(current_student_results[question.name])
+                for key, f in self.student_results_transforms():
+                    if key in question_results:
+                        question_results[key] = f(question_results, question)
```

Comment:
Nit: 

I would personally find this code easier to follow if you replace the above `for` loop with
```python
if 'choices' in question_results:
    question_results['choices'] = self.transform_choices(question_results, question)
if 'tip' in question_results:
    question_results['tips'] = self.transform_tips(question)
```
Which would also remove 9 lines of code. You could also change `transform_choices` so it won't throw an error if `result['choices']` is not a valid key, and then you can get rid of the `if` statements and make this only two lines.